### PR TITLE
Fix backup reminder shortcuts

### PR DESCRIPTION
**Pull request by BrodyHughes**
Fixes BX-1378

## What changed (plus any additional context for devs)
- this uses the `isExplainerSheet` class to determine whether or not the global shortcuts should be active
- while on an modal that `isExplainerSheet`, the `.`, `x`. `,`, etc shortcuts won't work. Only the normal navigation and `esc`. When the backup reminder is dismissed these will work again.
- this also enables the use of `esc` to close explainers

## Screen recordings / screenshots
<video src="https://github.com/rainbow-me/browser-extension/assets/41711440/ddea1c82-0f50-4393-8087-6719a23519fb" > </video>

## What to test
- `ESC` and `TAB` through items on the backup reminder
- attempt to use the `.` `,` `s` global shortcuts as well (they shouldn't work now while the backup reminder or explainer modals are up)

### CODE REVIEW

**Comment by BrodyHughes** [comment]
@DanielSinclair want some product feedback here. There are some implications on whether we want this to impact other modals (like token expanded state) as well as the backup reminders.

**Comment by DanielSinclair** [comment]
@DanielSinclair Just targeting the bottom-presenting modals like Backup Reminders should suffice. The issue was more so that shortcuts and actions were occurring invisibly behind the modal UI. We would still want the global shortcuts like `,` to work when navigating throughout layer 1 UI (where everything is presented with a left-right slide-in).

**Comment by DanielSinclair** [comment]
@BrodyHughes Just targeting the bottom-presenting modals like Backup Reminders should suffice. The issue was more so that shortcuts and actions were occurring invisibly behind the modal UI. We would still want the global shortcuts like , to work when navigating throughout layer 1 UI (where everything is presented with a left-right slide-in).

File: src/entries/popup/hooks/useHomeShortcuts.ts [hunk 1]
```diff
@@ -104,7 +105,9 @@ export function useHomeShortcuts() {
       const activeAppWalletSwitcher =
         appConnectionSwitchWalletsPromptIsActive();
       const inputIsFocused = getInputIsFocused();
+      const isModal = getActiveModal();
       if (inputIsFocused) return;
+      if (isModal) return;
```
Comment (DanielSinclair) [inline src/entries/popup/hooks/useHomeShortcuts.ts:110]: Poorly named I guess, but these are actually the global shortcuts here. We'd expect `w` to open the wallet switcher from any of the layer 1 drill-downs

**Review by DanielSinclair — CHANGES_REQUESTED** [review]

File: src/entries/popup/hooks/useHomeShortcuts.ts (hunk 1 above)
Comment (BrodyHughes) [reply src/entries/popup/hooks/useHomeShortcuts.ts:110]: ok cool cool. I can see what I can do in that case.

**Review by DanielSinclair — CHANGES_REQUESTED** [review]
Seeing some strange behavior for the ESC key in other parts of the extension now. If you go to token details, press tab, then esc, you'll see a stack of explainer sheets get triggered
Screenshot 2024-04-08 at 4 57 53 PM

**Comment by BrodyHughes** [comment]
> Seeing some strange behavior for the ESC key in other parts of the extension now. If you go to token details, press tab, then esc, you'll see a stack of explainer sheets get triggered Screenshot 2024-04-08 at 4 57 53 PM

ight fixed this. there was something weird going on with how we were using useReducer to change the boolean value. i just converted to useState and it works fine. useReducer is pretty unnecessary here anyways.

**Review by DanielSinclair — APPROVED** [review]
Looks good to me!

**Review by estebanmino — APPROVED** [review]

**Comment by DanielSinclair** [comment]
@derHowie if you have a sec to do a code review

**Review by derHowie — APPROVED** [review]
💯

**Pull request merged**